lit_gpt/utils.py

```diff
@@ -395,6 +395,15 @@ def CLI(*args: Any, **kwargs: Any) -> Any:
     return CLI(*args, **kwargs)
 
 
+def save_hyperparameters(function: callable, checkpoint_dir: Path) -> None:
+    """Captures the CLI parameters passed to `function` without running `function` and saves them to the checkpoint."""
+    from jsonargparse import capture_parser
+
+    parser = capture_parser(lambda: CLI(function))
+    config = parser.parse_args()
+    parser.save(config, checkpoint_dir / "hyperparameters.yaml", overwrite=True)
```

Comment:
Is it possible that there's an existing yaml with different values? Is it safe to overwrite it in that case?

Reply:
Yes, but then it means there is also a different `lit_model.pth` and gets overwritten. 
Whether we want to allow to overwrite checkpoints is something that can be discussed and boilerplate code be added to check it and so on. But in the PR I'm just following what we have done so far.